packages/transformer/src/IModelTransformer.ts

```diff
@@ -411,18 +414,144 @@ export class IModelTransformer extends IModelExportHandler {
+   * @param dbToQuery db to run the query on for scope external source
+   * @param aspectProps aspectProps to search for
+   * @param opts opts containing preserveVersionReceived and getJsonProperties. if preserveVersionReceived is true and an aspect is found with the provided aspectProps,
+   * then the version passed as part of the aspectProps will be kept on the aspectProps.
+   * If false or not provided, the version on the aspectProps will be overwritten with the version from the queried aspect.
+   */
+  public static queryScopeExternalSource(
+    dbToQuery: IModelDb,
+    aspectProps: ExternalSourceAspectProps,
+    opts?: { getJsonProperties?: boolean; preserveVersionReceived?: boolean }
+  ): void {
+    const sql = `
+      SELECT ECInstanceId, Version
+      ${opts?.getJsonProperties ? ", JsonProperties" : ""}
+      FROM ${ExternalSourceAspect.classFullName}
+      WHERE Element.Id=:elementId
+        AND Scope.Id=:scopeId
+        AND Kind=:kind
+        AND Identifier=:identifier
+      LIMIT 1
+    `;
+    dbToQuery.withPreparedStatement(sql, (statement: ECSqlStatement) => {
+      statement.bindId("elementId", aspectProps.element.id);
+      if (aspectProps.scope === undefined) return; // return instead of binding an invalid id
+      statement.bindId("scopeId", aspectProps.scope.id);
+      statement.bindString("kind", aspectProps.kind);
+      statement.bindString("identifier", aspectProps.identifier);
+      if (DbResult.BE_SQLITE_ROW !== statement.step()) return;
+      aspectProps.id = statement.getValue(0).getId();
+      aspectProps.version = opts?.preserveVersionReceived
+        ? aspectProps.version
+        : statement.getValue(1).getString();
+      aspectProps.jsonProperties = opts?.getJsonProperties
+        ? JSON.parse(statement.getValue(2).getString())
+        : undefined;
+    });
   }
 
-  private get _isForwardSynchronization() {
-    return this._isSynchronization && !this._options.isReverseSynchronization;
+  /**
+   * Determines the sync direction "forward" or "reverse" of a given sourceDb and targetDb by looking for the scoping ESA.
+   * If the sourceDb's iModelId is found as the identifier of the expected scoping ESA in the targetDb, then it is a forward synchronization.
+   * If the targetDb's iModelId is found as the identifier of the expected scoping ESA in the sourceDb, then it is a reverse synchronization.
+   * @throws if no scoping ESA can be found in either the sourceDb or targetDb which describes a master branch relationship between the two databases.
+   * @param sourceDb
+   * @param targetDb
+   * @param targetScopeElementId the element Id defined on the ExternalSourceAspect's scope. @see [[ExternalSourceAspectProps]]
+   * @returns "forward" or "reverse"
+   */
+  public static determineSyncType(
+    sourceDb: IModelDb,
+    targetDb: IModelDb,
+    targetScopeElementId: Id64String
```

Comment:
nit:
```suggestion
   * @returns "forward" or "reverse"
   */
  public static determineSyncType(
    sourceDb: IModelDb,
    targetDb: IModelDb,
    /** @see [[IModelTransformOptions.targetScopeElement]] */
    targetScopeElementId: Id64String
```
this is purely stylistic, resolve as you see fit. But imo it's more consistent when there are edges cases like inline parameter types

Also please double check my reference to `IModelTransformOptions.targetScopeElement`, but I think centralizing that definition in the docs is better.